src/Datasource/EntityTrait.php

```diff
@@ -787,6 +787,43 @@ public function getDirty()
     }
 
     /**
+     * return value if property exists otherwise throw Exception
+     *
+     * @param string $property The field to get value for.
+     * @return mixed
+     * @throws \InvalidArgumentException if the property name doesn't exist
+     */
+    public function &getOrFail($property)
+    {
+        $value = $this->get($property);
+        if (!$this->propertyExists($property) && $value === null) {
```

Comment:
If you invert these two, it would be a bit more performent in same cases. Nano-opt. :)

Reply:
Oh I missed this one - usually I take care of the order. Thanks for your feedback ;-)